neurolib/optimize/exploration/explorationUtils.py

```diff
@@ -36,79 +46,119 @@ def plotExplorationResults(
         par2_label = par2[1]
         par2 = par2[0]
 
+    # the `by` argument is used to slice the data
     # if no by value was given, create a dummy value
     if by is None:
         dfResults["_by"] = 0
         by = ["_by"]
 
-    n_plots = len(dfResults.groupby(by=by))
-
     if by_label is None:
-        by_label = by
+        by_label = by        
 
-    # if by is not None:
-    #     n_plots = len(dfResults.groupby(by=by))
-    # else:
-    #     n_plots = 1
+    n_plots = len(dfResults.groupby(by=by))
+
+    # PLOT
+    # ------------------    
 
+    # create subfigures
     if one_figure == True:
         fig, axs = plt.subplots(nrows=1, ncols=n_plots, figsize=(n_plots * 4, 3.5), dpi=150)
-
-    # if by is not None:
-    #     _dfResults = dfResults.groupby(by=by)
-    # else:
-    #     _dfResults = dfResults
+        if plot_key_label:
+            fig.suptitle(plot_key_label)
 
     axi = 0
+    # cycle through all slices
     for i, df in dfResults.groupby(by=by):
-
+        # chose the current axis
         if one_figure == True:
             if n_plots > 1:
                 ax = axs[axi]
             else:
                 ax = axs
+        else:
+            fig = plt.figure(figsize=(5, 4), dpi=150)
+            if plot_key_label:
+                plt.title(plot_key_label)
+            ax = plt.gca()       
 
-        df_pivot = df.pivot_table(values=plot_key, index=par2, columns=par1)
 
-        plot_clim = None
-        if symmetric_colorbar:
-            plot_clim = (-np.max(df_pivot.values), np.max(df_pivot.values))
+        # -----
+        # pivot data and plot
+        df_pivot = df.pivot_table(values=plot_key, index=par2, columns=par1, dropna=False)
 
-        if one_figure == False:
-            fig = plt.figure(figsize=(5, 4), dpi=150)
-            ax = plt.gca()
+        plot_clim = (np.nanmin(df_pivot.values), np.nanmax(df_pivot.values))
+
+        if symmetric_colorbar:
+            plot_clim = (-np.nanmax(df_pivot.values), np.nanmax(df_pivot.values))
```

Comment:
what if abs(min_value) > max_value? I think better is

```suggestion
            plot_clim = (-np.max(np.abs(plot_clim)), np.max(np.abs(plot_clim)))
```